tests/functional/simple_copy/test_simple_copy.py

```diff
@@ -0,0 +1,217 @@
+import pytest
+import os
+from tests.util import run_dbt, copy_file, run_sql
+from tests.tables import TableComparison, get_tables_in_schema
+
+# advanced_incremental.sql
+advanced_incremental_sql = """
+{{
+  config(
+    materialized = "incremental",
+    unique_key = "id",
+    persist_docs = {"relation": true}
+  )
+}}
+
+select *
+from {{ ref('seed') }}
+
+{% if is_incremental() %}
+
+    where id > (select max(id) from {{this}})
+
+{% endif %}
+"""
+
+# compound_sort.sql
+compound_sort_sql = """
+{{
+  config(
+    materialized = "table",
+    sort = 'first_name',
+    sort_type = 'compound'
+  )
+}}
+
+select * from {{ ref('seed') }}
+"""
+
+# disabled.sql
+disabled_sql = """
+{{
+  config(
+    materialized = "view",
+    enabled = False
+  )
+}}
+
+select * from {{ ref('seed') }}
+"""
+
+# empty.sql
+empty_sql = """
+"""
+
+
+# get_and_ref.sql
+get_and_ref_sql = """
+{%- do adapter.get_relation(database=target.database, schema=target.schema, identifier='materialized') -%}
+
+select * from {{ ref('materialized') }}
+"""
+
+# incremental.sql
+incremental_sql = """
+{{
+  config(
+    materialized = "incremental"
+  )
+}}
+
+select * from {{ ref('seed') }}
+
+{% if is_incremental() %}
+    where id > (select max(id) from {{this}})
+{% endif %}
+"""
+
+# interleaved_sort.sql
+interleaved_sort_sql = """
+{{
+  config(
+    materialized = "table",
+    sort = ['first_name', 'last_name'],
+    sort_type = 'interleaved'
+  )
+}}
+
+select * from {{ ref('seed') }}
+"""
+
+# materialized.sql
+materialized_sql = """
+{{
+  config(
+    materialized = "table"
+  )
+}}
+-- ensure that dbt_utils' relation check will work
+{% set relation = ref('seed') %}
+{%- if not (relation is mapping and relation.get('metadata', {}).get('type', '').endswith('Relation')) -%}
+    {%- do exceptions.raise_compiler_error("Macro " ~ macro ~ " expected a Relation but received the value: " ~ relation) -%}
+{%- endif -%}
+-- this is a unicode character: å
+select * from {{ relation }}
+"""
+
+# schema.yml
+schema_yml = """
+version: 2
+models:
+- name: disabled
+  columns:
+  - name: id
+    tests:
+    - unique
+"""
+
+# view_model.sql
+view_model_sql = """
+{{
+  config(
+    materialized = "view"
+  )
+}}
+
+select * from {{ ref('seed') }}
+"""
+
+
+@pytest.fixture
+def models():
+    return {
+        'advanced_incremental.sql': advanced_incremental_sql,
+        'compound_sort.sql': compound_sort_sql,
+        'disabled.sql': disabled_sql,
+        'empty.sql': empty_sql,
+        'get_and_ref.sql': get_and_ref_sql,
+        'incremental.sql': incremental_sql,
+        'interleaved_sort.sql': interleaved_sort_sql,
+        'materialized.sql': materialized_sql,
+        'schema.yml': schema_yml,
+        'view_model.sql': view_model_sql,
```

Comment:
Should we aim to consolidate these fixtures with the ones used in other tests? It would require more upfront work, but if we assembled a common set of standard fixtures (rather than recreating highly similar ones over and over again), then:
- each test can pick & choose just the ones it needs
- adapter plugins could reimplement specific fixtures (as needed) among the standard reused set, and then be able to get most tests "for free," rather than needing to reimplement similar fixtures in similar ways for each test

Reply:
Ideally I think it would be nice to have some standard projects that can be reused. We have a lot of tests that create projects with only subtle or minimal differences. That can be done a directory basis, using a test directory conftest.py, with only the different parts overridden. We probably could do that for test_simple_copy.py and test_copy_upperbase, and just override the 'models' fixture for test_copy_upperbase. You can also make individual sets of project fixtures with different names and include them in a test as a plugin.

I think it will be easiest to do that after we've converted the tests though, since we'll then be able to more easily see the differences between test projects. The legacy tests with all the files in directories (often multiple) makes that sort of comparison harder.

Reply:
Though on second thought, if people see the same setup as they're going through converting, there's no reason to wait. "Oh, I saw this project before. Yank! It's now a plugin."